sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/CSVSink.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.snowflake;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.nio.charset.Charset;
+import java.util.List;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Joiner;
+
+/** Implementation of {@link org.apache.beam.sdk.io.FileIO.Sink} for writing CSV. */
+public class CSVSink implements FileIO.Sink<String> {
```

Comment:
I took only a very superficial look. Are you using CSV file imports to write to Snowflake? I would be concerned about types being properly encoded in the CSV (big integers, floats, bytes, date/time types, strings with newlines/quotes). Have you already considered how to deal with all types?

Reply:
Yes, we are using [COPY INTO table](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#type-csv) with CSV format. 

`SnowflakeIO.write `is constructed in a way that requires a particular table to exist in Snowflake before starting writing into Snowflake.

Reply:
Thanks for spotting this. I removed `CSVSink` class and start using `.via(TextIO.sink())` .

Reply:
In one of the PR's , we're planing to add `create dispositions` for tables  which is connected more with schema concept. Let's talk about it then :)